src/main/java/backtrace/io/BacktraceQueue.java

```diff
@@ -33,32 +40,46 @@ void addWithLock(BacktraceMessage message) {
     void unlock() {
         LOGGER.debug("Releasing semaphore..");
 
-        if(notEmptyQueue.getCount() == 0) {
-            notEmptyQueue.countUp();
+        if (processingLock.getCount() == 1) {
+            processingLock.countDown();
         }
+    }
+
+    boolean shouldHandleMessages() {
+        return processingLock.getCount() == 1;
+    }
 
-        if (lock.getCount() == 0) {
-            return;
+    /**
+     * Lock semaphore because queue is empty
+     */
+    void queueIsEmpty() {
```

Comment:
proposal:
`queueIsEmtpy` => `lockQueue` 
`queueNotEmpty` => `unlockQueue` 

For example: `queueIsEmtpy` doesn't tell you what the function does.

Reply:
`queueIsEmpty` it is a logic which should be run when queue is empty